packages/schema/src/Schema.ts

```diff
@@ -2023,111 +2086,136 @@ export interface TypeLiteral<
+          issTo.push(new AST.IndexSignature(is.parameter, AST.typeAST(is.type), is.isReadonly))
+        })
+      }
+      return new AST.Transformation(
+        new AST.TypeLiteral(from, issFrom, { [AST.TitleAnnotationId]: "Struct (Encoded side)" }),
+        new AST.TypeLiteral(to, issTo, { [AST.TitleAnnotationId]: "Struct (Type side)" }),
+        new AST.TypeLiteralTransformation(transformations)
+      )
     }
-    return new AST.TypeLiteral(pss, iss)
   }
-  readonly fields: { readonly [K in keyof Fields]: Fields[K] }
-  readonly records: Readonly<Records>
-  constructor(
-    fields: Fields,
-    records: Records,
-    ast: AST.AST = TypeLiteralImpl.ast(fields, records)
-  ) {
-    super(ast)
-    this.fields = { ...fields }
-    this.records = [...records] as Records
+  const iss: Array<AST.IndexSignature> = []
+  for (const r of records) {
+    const { indexSignatures, propertySignatures } = AST.record(r.key.ast, r.value.ast)
+    propertySignatures.forEach((ps) => pss.push(ps))
+    indexSignatures.forEach((is) => iss.push(is))
   }
-  annotations(
-    annotations: Annotations.Schema<Types.Simplify<TypeLiteral.Type<Fields, Records>>>
-  ): TypeLiteral<Fields, Records> {
-    return new TypeLiteralImpl(this.fields, this.records, AST.annotations(this.ast, toASTAnnotations(annotations)))
+  return new AST.TypeLiteral(pss, iss)
+}
+
+const getFieldsDefaults = (fields: Struct.Fields): object => {
+  const out: any = {}
+  const ownKeys = util_.ownKeys(fields)
+  for (const key of ownKeys) {
+    const field = fields[key]
+    if (isPropertySignature(field)) {
+      const ast = field.ast
+      const defaultValue = ast._tag === "PropertySignatureDeclaration" ? ast.defaultValue : ast.to.defaultValue
+      if (defaultValue !== undefined) {
+        out[key] = defaultValue()
+      }
+    }
+  }
+  return out
+}
+
+const makeTypeLiteralClass = <
+  Fields extends Struct.Fields,
+  const Records extends IndexSignature.Records
+>(
+  fields: Fields,
+  records: Records,
+  ast: AST.AST = getDefaultTypeLiteralAST(fields, records)
+): TypeLiteral<Fields, Records> => {
+  return class TypeLiteralClass extends make<
+    Types.Simplify<TypeLiteral.Type<Fields, Records>>,
+    Types.Simplify<TypeLiteral.Encoded<Fields, Records>>,
+    | Struct.Context<Fields>
+    | IndexSignature.Context<Records>
+  >(ast) {
+    static override annotations(
+      annotations: Annotations.Schema<Types.Simplify<TypeLiteral.Type<Fields, Records>>>
+    ): TypeLiteral<Fields, Records> {
+      return makeTypeLiteralClass(this.fields, this.records, AST.annotations(this.ast, toASTAnnotations(annotations)))
+    }
+
+    static fields = { ...fields }
+
+    static records = [...records] as Records
+
+    static make(
+      props: Types.Simplify<TypeLiteral.Constructor<Fields, Records>>
+    ): Types.Simplify<TypeLiteral.Type<Fields, Records>> {
+      return ParseResult.validateSync(this)({ ...getFieldsDefaults(fields), ...props as any })
```

Comment:
better, but I wonder how expensive this is getting for calls that do set them,
like they're lazy but are activated every time a new instance is created, regardless if they will be discarded immediately after. for Date objects it may become expensive.